forge-game/src/main/java/forge/game/card/CardView.java

```diff
@@ -40,6 +41,14 @@ public static CardStateView getState(Card c, CardStateName state) {
         return s == null ? null : s.getView();
     }
 
+    public static Map<CardStateView, CardState> getStateMap(Iterable<CardState> states) {
+        Map<CardStateView, CardState> stateViewCache = Maps.newLinkedHashMap();
+        for (CardState state : states) {
+            stateViewCache.put(state.getView(), state);
```

Comment:
this doesn't work yet because (not sure yet if bug or not) both cardstates from room have same id
so map never gets both entries :/

Reply:
should be fixed?